### PR TITLE
bpo-33522: Enable CI builds on Visual Studio Team Services

### PR DESCRIPTION
**Pull request by zooba**
We have been provided with https://python.visualstudio.com/cpython and a measure of build machines to help us run continuous integration builds. The benefits are more reliable builds (especially Mac and Windows) and higher parallelism than our current services.

This PR adds the build definition files and fixes some test bugs to ensure we can successfully test on this platform. There are a few more manual steps to make it work against our repo, which we will do immediately after merging the changes.

There are also some more changes coming, but feel free to start the review :)

https://bugs.python.org/issue33522

### CODE REVIEW

File: .vsts/docs.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+queue:
+  name: Hosted Linux Preview
+
+trigger:
+  branches:
+    include:
+    - master
+    - 3.7
+    - 3.6
+    - vsts
```
Comment (brettcannon) [inline .vsts/docs.yml]: I think this branch is no longer necessary.

File: Lib/test/support/__init__.py [hunk 2]
```diff
@@ -363,6 +363,20 @@ def _rmtree_inner(path):
                     _force_run(fullname, os.unlink, fullname)
         _waitfor(_rmtree_inner, path, waitall=True)
         _waitfor(lambda p: _force_run(p, os.rmdir, p), path)
+
+    def _longpath(path):
+        try:
+            import ctypes
+        except ImportError:
+            # No ctypes means we can't expands paths
```
Comment (brettcannon) [inline Lib/test/support/__init__.py]: Missing a period.

**Review by brettcannon — APPROVED** [review]

**Comment by gpshead** [comment]
The test and standard library changes should not have been part of this PR.  Now those have been committed, including backports to two stable releases, with no description of what changed in the commit message and inadequate news entries.

The Library NEWS entry fails to mention that the change was Windows specific.

The test.support Library change isn't even documented with a NEWS entry.

There is no description of what changed at all in the commit message.  The commit message is the most important place to record such information.  It is the _only_ thing we have on permanent record.  This PR and Github are fly-by-night transient sources of information that can and will disappear in the future.

@nedbat

**Comment by gpshead** [comment]
@ned-deily

**Comment by ned-deily** [comment]
There *are* a lot of changes in this PR and, AFAICT, there are changes to non-Windows tests.  Do we know if any of the changes correct problems already documented in other b.p.o issues?   I agree with @gpshead that such a pervasive change, especially coming in at the last minute, should be better documented (PRs are not sufficient).  Downstream users want to know what has changed between, say, 3.7.0b4 and 3.7.0rc1 without having to resort to exhaustive diffs; that why we have NEWS entries and the changelog html.

**Pull request merged**